src/GitHubActions.jl

```diff
@@ -188,10 +190,12 @@ Set environment variable `k` to value `v`.
 function set_env(k, v)
     val = cmd_value(v)
     ENV[k] = val
```

Comment:
Setting `ENV` should probably be moved to have the safety-precaution checks, imo. I guess it doesn't matter if we're literally going to exit, but if we refactored that to throw or something instead, then we'd have a bug